swarm/storage/dbstore.go

```diff
@@ -440,6 +469,20 @@ func (s *DbStore) Put(chunk *Chunk) {
 	log.Trace(fmt.Sprintf("DbStore.Put: %v. db storage counter: %v ", chunk.Key.Log(), s.dataIdx))
 }
 
+func (s *DbStore) dbPutDataFunc(batch *leveldb.Batch, _ Key, data []byte) {
```

Comment:
do I see it correctly that with this it wont be possible to iterate over chunks by peer by bin without the datakey index in the node's dbStore. This is avoided if we just use encode/decode right?

Reply:
This is related to the previous comment, but saving the data index is with nil data for iteration could be done by adding bucket.Put to the function that handles the mock store Put. But yes, extending the functionality of encodeData would solve the issue.